libbeat/docs/shared-docker.asciidoc

```diff
@@ -100,11 +100,24 @@ configuration included.
 
 Download this example configuration file as a starting point:
 
+ifndef::apm-server[]
+
 ["source","sh",subs="attributes,callouts"]
 ------------------------------------------------
 curl -L -O https://raw.githubusercontent.com/elastic/beats/{doc-branch}/deploy/docker/{beatname_lc}.docker.yml
```

Comment:
Rather than using an ifndef here, I'd recommend using an attribute to resolve the path (see the attributes that already exist for `dockerimage` and `dockerimage`). For beats, you can define the attributes in `libbeat/docs/shared-beats-attributes`, and all the Beats books will pick it up. Looks like you're not using the Beats shared attributes file, so you'll have to define the attribute in your index file.

I still have a to-do to cleanup and document how our attributes are used.

Reply:
I'm not sure I understand. I can see where you've used `dockerimage` earlier in the file, but it seems to be unrelated to this example? Let's chat about this when you get back from vacation! And actually, I'd like to walk through this whole PR with you when you return just to clarify a few things if that's OK.

Reply:
If you want, we can revisit this issue when we clean up the conditional coding. The gist is that you we can use an asiidoc attribute in places to resolve paths rather than using ifndef/ifdef.

Reply:
Ahh, so are you suggesting I create an attribute named, for example: `{docker_download}`. Then set it in two places?

```
// shared-beats-attributes
:docker_download: curl -L -O https://raw.githubusercontent.com/elastic/beats/{doc-branch}/deploy/docker/{beatname_lc}.docker.yml
// apm-server/docs/index.asciidoc
:docker_download: curl -L -O https://raw.githubusercontent.com/elastic/apm-server/{doc-branch}/apm-server.docker.yml

Reply:
Alright, ended up adding in `{dockerconfig}` as an attribute.